vm/src/org/mozartoz/truffle/nodes/builtins/NumberBuiltins.java

```diff
@@ -61,6 +64,52 @@ long neg(long value) {
 
 	}
 
+	@Builtin(name = "pow", deref = ALL)
+	@GenerateNodeFactory
+	@NodeChildren({ @NodeChild("left"), @NodeChild("right") })
+	public static abstract class PowNode extends OzNode {
+
+		@Specialization(guards = { "isInteger(base)", "exponent <= 94", "exponent == cachedExponent" })
```

Comment:
94? That's big :p
One optimization you can do is put the guard on the `cachedExponent`, so it's only executed once :)